### DISCUSSION TITLE
How to cite OpenCilk and its documentation

### DISCUSSION THREAD

**Discussion by behoppe** [OpenCilk/opencilk-project#174] [Q&A]
@adnan4791 asks a good question: how to cite the [Cilkscale reference page](https://www.opencilk.org/doc/reference/cilkscale/), and I'm posting for discussion because I'd like to generalize the question and also hear team input.

To cite OpenCilk documentation, I would follow guidelines [like this](https://owl.purdue.edu/owl/research_and_citation/mla_style/mla_formatting_and_style_guide/mla_works_cited_electronic_sources.html) for citing a website. Although he got lots of input from the team, I think it's fair to say that @ailiop is the author of the Cilkscale reference article.

To cite OpenCilk, it depends, and I have a little homework to do on this. In earlier Cilk documentation (no longer online), there was this little guide on "what to cite" with different references for each of the following features of OpenCilk:

* The DAG model of multithreading (BlumofeLe98)
* Randomized work stealing (BlumofeLe99)
* The Cilk language (i.e., `spawn` and `sync`) (FrigoLeRa98)
* Cilk reducer hyperobjects (FrigoHeLe09)
* Determinacy-race detection (FengLe99)
* The Cilkview scalability analyzer (HeLeLe10)
* The Cilkprof scalability profiler (ShardlKeLe15)
* The Tapir/LLVM compiler (SchardlMoLe17)

Adnan, I think the last three might be relevant to you. I will follow up soon with more info on the "encoded" references above (and integrate them into our current OpenCilk website). 

@neboat would you add or change anything on this list? Is there a single recommended reference for Cilkscale (as we have for Cilkview and Cilkprof)?

**Comment by adnan4791** [comment]
Hi Bruce 

thank you,  your post and answer help me much
